src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt

```diff
@@ -76,4 +76,21 @@ class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
 
         return jpaQuery.resultList.map { it as Array<Any> }.map { Pair(it[0] as HearingEntity, it[1] as LocalDate) };
     }
+
+    fun getDynamicOutcomeCountsByState(courtCode: String): Map<String, Int> {
```

Comment:
same here, I can see the state is involved in the query but struggling to understand the relevance of it in the name. please clarify.